lib/puppet/feature/iis_compatible_version_installed.rb

```diff
@@ -0,0 +1,43 @@
+require 'puppet/util/feature'
+
+# Consider joining MajorVersion + MinorVersion instead of using VersionString
+
+def read_registry_for_iis_compatible_version
+  compatible_versions = ['7.5', '8.0', '8.5', '10.0']
+  installed_version = '0.0'
+
+  Puppet.debug "Reading Registry for IIS Version"
+
+  begin
+    require 'win32/registry'
+
+    f_HKLM        = Win32::Registry::HKEY_LOCAL_MACHINE
+    f_REG_PATH    = 'SOFTWARE\Microsoft\InetStp'
+    f_ACCESS_TYPE = Win32::Registry::KEY_READ | 0x100
+    f_REG_KEY     = 'VersionString'
+
+    f_HKLM.open(f_REG_PATH, f_ACCESS_TYPE) do |reg|
+      installed_version = reg[f_REG_KEY][8,3]
```

Comment:
I don't think this works for `10.0` given it's 4 characters long.

Instead, I think you can do something like:

```ruby
installed_version = Gem::Version.new(reg[f_REG_KEY][8..-1]).segments[0, 2].join('.')
```

Reply:
Addressed via changes in MODULES-5801
See also: lib/puppet/provider/templates/webadministration/logproperties.ps1.erb